### PR TITLE
Evaluate conditional annotations lazily to avoid VM bootstrap issues

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

File: plugins/patcher/src/main/java/org/qbicc/plugin/patcher/MethodDeleteInfo.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        return descriptor;
+    }
+
+    Annotation getAnnotation() {
+        return annotation;
+    }
+
+    @Override
+    public Location getLocation() {
+        return ClassContextPatchInfo.getMethodLocation(internalName, "<init>");
```
Comment (dgrove-oss) [inline plugins/patcher/src/main/java/org/qbicc/plugin/patcher/MethodDeleteInfo.java:35]:
```suggestion
        return ClassContextPatchInfo.getMethodLocation(internalName, name);
```

Comment (dgrove-oss) [reply plugins/patcher/src/main/java/org/qbicc/plugin/patcher/MethodDeleteInfo.java:35]: Looks like this is a copy & paste mistake from ConstructorDeleteInfo??

**Review by dgrove-oss — APPROVED** [review]
Looks good modulo what I think is a cut&paste error

File: plugins/patcher/src/main/java/org/qbicc/plugin/patcher/MethodDeleteInfo.java (hunk 1 above)
Comment (dmlloyd) [reply plugins/patcher/src/main/java/org/qbicc/plugin/patcher/MethodDeleteInfo.java:35]: Good catch...

**Pull request merged**